encryption/reader.go

```diff
@@ -0,0 +1,36 @@
+package encryption
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"io"
+)
+
+type StreamDecrypter struct {
+	Source io.Reader
+	Block  cipher.Block
+	Stream cipher.Stream
+}
+
+func (s *StreamDecrypter) Read(p []byte) (int, error) {
+	n, readErr := s.Source.Read(p)
+	if n > 0 {
+		s.Stream.XORKeyStream(p[:n], p[:n])
```

Comment:
https://golang.org/src/crypto/cipher/io.go?s=934:1010#L20
_A StreamWriter has no internal buffering;_

`cipher.NewCTR` uses 512 bytes for buffering, this is super ineffective for transferring large amounts of data